src/kernel/boot.c

```diff
@@ -693,6 +711,19 @@ BOOT_CODE static bool_t provide_untyped_cap(
             .isDevice = device_memory,
             .padding  = {0}
         };
+
+#if defined(__CHERI_PURE_CAPABILITY__)
+        if (!device_memory) {
+            /*
+             * Make sure to apply proper bounds on new untyped memory. The received pptr could have winder
+             * bounds from the caller as it tries to allocate multiple power-of-two untypes from a single
+             * bigger chunk of memory. We only build caps for non-device memory as that's what the kernel
+             * could acces; device memory isn't accessed by the kernel.
+             */
+            pptr = (pptr_t) cheri_derive_data_cap((void *)pptr, (ptraddr_t)pptr, BIT(size_bits), -1);
+        }
+#endif
```

Comment:
How is user space given access to the memory if the kernel has no right to it?

What about cache operations, do they need access too?

Maybe the answer to both is that the rights are gotten via another way, but worth a comment perhaps.

Reply:
This change _only_ builds a CHERI capability for the kernel for this untype so that the kernel can reference it later (e.g.,, zeroing it).

Cache operations for user addresses do need access, yes, there's a user cap built for it here https://github.com/CTSRD-CHERI/seL4/blob/cheri/src/arch/arm/64/kernel/vspace.c#L1712